tests/tasks/test_bgpstatemonitortask.py

```diff
@@ -111,18 +292,103 @@ async def test_peer_admin_status_changing_to_stop_while_peer_state_is_not_establ
         task = BgpStateMonitorTask(device, state)
         peer_address = IPv4Address("10.0.0.2")
         # set initial state
-        task.device_state.bgp_peer_admin_states = {peer_address: "start"}
+        task.device_state.bgp_peer_admin_states = {peer_address: "running"}
         await task.run()
         # check if state has been updated to reflect state defined in .snmprec
         assert task.device_state.bgp_peer_admin_states[peer_address] == "halted"
         assert task.device_state.bgp_peer_oper_states[peer_address] != "established"
+        # check that the correct event has been created
+        event = task.state.events.get(device_name=task.device.name, port=peer_address, event_class=BGPEvent)
+        assert event
+        assert event.admin_status == "halted"
+        assert event.operational_state == "down"
+        assert event.remote_address == peer_address
+        assert event.remote_as == 10
+        assert event.peer_uptime == 0
+
+    @pytest.mark.asyncio
+    async def test_admin_up_general_creates_event(self, snmpsim, snmp_test_port):
+        device = PollDevice(
+            name=DEVICE_NAME,
+            address=DEVICE_ADDRESS,
+            community="general-bgp-admin-up",
+            port=snmp_test_port,
+        )
+        state = ZinoState()
+        task = BgpStateMonitorTask(device, state)
+        peer_address = IPv4Address("10.0.0.1")
+        # set initial state
+        task.device_state.bgp_peer_admin_states = {peer_address: "stop"}
+        task.device_state.bgp_peer_oper_states = {peer_address: "idle"}
+        await task.run()
+        # check if state has been updated to reflect state defined in .snmprec
+        assert task.device_state.bgp_peer_admin_states[peer_address] == "start"
+        assert task.device_state.bgp_peer_oper_states[peer_address] != "established"
+        # check that the correct event has been created
         event = task.state.events.get(device_name=task.device.name, port=peer_address, event_class=BGPEvent)
         assert event
+        assert event.admin_status == "start"
+        assert event.operational_state == "idle"
+        assert event.remote_address == peer_address
+        assert event.remote_as == 20
+        assert event.peer_uptime == 0
 
     @pytest.mark.asyncio
-    async def test_remote_as_different_from_local_as_changing_oper_state_to_down_should_create_event_general(
-        self, snmpsim, snmp_test_port
-    ):
+    async def test_admin_up_cisco_creates_event(self, snmpsim, snmp_test_port):
+        device = PollDevice(
+            name=DEVICE_NAME,
+            address=DEVICE_ADDRESS,
+            community="cisco-bgp-admin-up",
+            port=snmp_test_port,
+        )
+        state = ZinoState()
+        task = BgpStateMonitorTask(device, state)
+        peer_address = IPv4Address("10.0.0.1")
```

Comment:
this is another candidate for "normalization". Cisco used `10.0.0.1` for `peer_adress` while juniper and general uses `10.0.0,2`